AsyncDisplayKit/Private/ASDisplayNode+AsyncDisplay.mm

```diff
@@ -83,6 +83,34 @@ - (NSObject *)drawParameters
   return nil;
 }
 
++ (void)drawRect:(CGRect)bounds withParameters:(id <_ASDisplayLayerDelegateForward>)parameters isCancelled:(asdisplaynode_iscancelled_block_t)isCancelledBlock isRasterizing:(BOOL)isRasterizing
+{
+  ASDisplayNodeAssert([parameters conformsToProtocol:@protocol(_ASDisplayLayerDelegateForward)], @"Calling super for drawRect:withParameters:isCancelled:isRasterizing: must pass in the original parameters to superclass implementation.");
+  
+  if ([parameters conformsToProtocol:@protocol(_ASDisplayLayerDelegateForward)]) {
+    id <_ASDisplayLayerDelegateForward> internalParameters = (id <_ASDisplayLayerDelegateForward>)parameters;
+    id<_ASDisplayLayerDelegate> delegate = internalParameters.delegate;
```

Comment:
@maicki this is quite a smart, creative implementation! I had not thought of a reasonable way to support a super call triggering an instance method, coming from a class method!

My only concern with this route is that it requires keeping the draw parameters, Specifically in the instance method case, which is where we would like to eliminate them. I am more than happy to run some up-to-date profiles to quantify the current cost of the parameters now that fewer elements are being put into the dictionaries.  This is definitely a solvable problem by perhaps adding just the delegate to the parameters, or something else like how we handled the layout context.

I would suggest we put this patch aside for now, because there are some pressing issues to prioritize for the 2.0 release. This would be fun to revisit in one of the follow-up releases, as refining the developer APIs for drawing definitely has a few other opportunities that we can investigate.

[sorry for the odd english, relying on voice dictation sucks]